packages/analytics/amplify_analytics_pinpoint_dart/test/analytics_legacy_native_data_test.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+import 'package:uuid/uuid.dart';
+
+void main() {
+  group('Analytics Legacy Native Data Tests', () {
+    late MockLegacyNativeDataProvider legacyDataProvider;
+    late MockSecureStorage store;
+
+    const appId = 'appId';
+    const legacyEndpointId = 'legacy-endpointId';
+
+    setUp(() {
+      legacyDataProvider = MockLegacyNativeDataProvider();
+      store = MockSecureStorage();
+    });
+
+    test('First app load, no legacy data, writes proper values', () async {
+      when(() => legacyDataProvider.getEndpointId(appId))
+          .thenAnswer((_) => Future.value());
+
+      await AmplifyAnalyticsPinpointDart.retrieveEndpointId(
+        pinpointAppId: appId,
+        store: store,
+        legacyDataProvider: legacyDataProvider,
+      );
+
+      final storeVersion = await store.read(
+        key: EndpointStoreKey.version.name,
+      );
+      expect(storeVersion, EndpointStoreVersion.v1.name);
+
+      final migratedEndpointId = await store.read(
+        key: AmplifyAnalyticsPinpointDart.endpointIdStorageKey,
+      );
+      expect(migratedEndpointId, isNotNull);
+
+      verify(() => legacyDataProvider.getEndpointId(appId)).called(1);
+    });
+
+    test('First app load, legacy data, writes proper values', () async {
+      when(() => legacyDataProvider.getEndpointId(appId))
+          .thenAnswer((_) => Future.value(legacyEndpointId));
+
+      await AmplifyAnalyticsPinpointDart.retrieveEndpointId(
+        pinpointAppId: appId,
+        store: store,
+        legacyDataProvider: legacyDataProvider,
+      );
+
+      final storeVersion = await store.read(
+        key: EndpointStoreKey.version.name,
+      );
+      expect(storeVersion, EndpointStoreVersion.v1.name);
+
+      final migratedEndpointId = await store.read(
+        key: AmplifyAnalyticsPinpointDart.endpointIdStorageKey,
+      );
+      expect(migratedEndpointId, legacyEndpointId);
+
+      verify(() => legacyDataProvider.getEndpointId(appId)).called(1);
+    });
+
+    test('Second app load, legacy data is ignored', () async {
+      const legacyEndpointId = 'legacy-endpointId';
+      when(() => legacyDataProvider.getEndpointId(appId))
+          .thenAnswer((_) => Future.value(legacyEndpointId));
+
+      final endpointId = const Uuid().v1();
+      store.seedData({
+        EndpointStoreKey.version.name: EndpointStoreVersion.v1.name,
+        AmplifyAnalyticsPinpointDart.endpointIdStorageKey: endpointId
+      });
+
+      await AmplifyAnalyticsPinpointDart.retrieveEndpointId(
+        pinpointAppId: appId,
+        store: store,
+        legacyDataProvider: legacyDataProvider,
+      );
+
+      final migratedEndpointId = await store.read(
+        key: AmplifyAnalyticsPinpointDart.endpointIdStorageKey,
+      );
+      expect(migratedEndpointId, endpointId);
+
+      verifyNever(() => legacyDataProvider.getEndpointId(captureAny()));
```

Comment:
*nit:* `captureAny()` is intended to match any argument _and_ save the value of the argument. Since you don't need to save the value, you can use `any()` which matches any argument, but doesn't save it.

Example use of `captureAny()` from the docs:
```dart
when(() => cat.likes('fish')).thenReturn(true);
expect(cat.likes('fish'), isTrue);
final captured = verify(() => cat.likes(captureAny())).captured;
expect(captured.last, equals(['fish']))
``` 

```suggestion
      verifyNever(() => legacyDataProvider.getEndpointId(any()));
```